### DISCUSSION TITLE
Where did v7 documentation go?

### DISCUSSION THREAD

**Discussion by ryanm2000** [TanStack/table#4885] [General] [answered]
We are unable to migrate from v7 to v8. As recently as 19th May 2023 the v7 documentation site was very useful. However, upon checking today, navigating to https://react-table-v7.tanstack.com/ seems to redirect to https://tanstack.com/table/v8/?from=reactTableV7&original=https://github.com/TanStack/table/tree/v7/docs/src/pages/ 

The latter is a far less helpful page, even when clicking the v7 docs button at the top of the screen, because it just goes to the github source of the project, which presumably need to be built and run locally for all who want to use it.

**Comment by ceoshikhar** [comment]
Yes, even I was looking for v7 docs and it just redirected me to Github. I really needed the docs and it's just sad I can't.

Can we get this fixed or is it what @tannerlinsley wanted so that people start using v8?

**Comment by tannerlinsley** [comment]
I’m moving v7 docs over to the new website. GitHub docs are the temporary location.

Tanner Linsley

**Reply by ceoshikhar** [reply]
Got it. Thanks for the update. 🚀

**Reply by ryanm2000** [reply]
Thanks for the update, Tanner. Appreciate all your great work!

**Reply by Ratandeep13** [reply]
@tannerlinsley Could you please share website link

**Reply by kevin-dev71** [reply]
@Ratandeep13 https://github.com/TanStack/table/tree/v7/docs

**Comment by DeepjyotiDeb** [comment]
Hello, any updates on the link to the docs for v7? not the github docs please. Thanks!

**Comment by targhs** [comment]
Docs were very useful. Its too difficult to navigate in github finding what you need. Any update?

**Reply by lgsf** [reply]
There's a way, using internet archive. Everything seems to be there.
https://web.archive.org/web/20220926134341/https://react-table-v7.tanstack.com/docs/examples/pagination

**Comment by lgsf** [comment]
I found a way, thanks to internet archive:

https://web.archive.org/web/20220926134341/https://react-table-v7.tanstack.com/docs/

**Reply by avshyz-gs** [reply]
Broken :(
Is there any other way to do that?

**Reply by mcookdev** [reply]
This site is often blocked by corporations.

**Reply by quantizor** [reply]
Yeah this isn't working anymore

**Comment by Josh68** [comment]
The internet archive no longer works for this. It redirects to a version of the docs that linked to the v7 docs, but now even that link redirects to the repo. The repo checked out on the v7 branch cannot be built, either. There's a dependency issue (CJS / ESM incompatibility, I think)

**Reply by Josh68** [reply]
I just did mange to get the docs running locally. Don't know what the issue was, exactly, but I used node gallium/lts (v16) and npm (not yarn), and it worked

**Reply by Josh68** [reply]
Also works with Node LTS (iron) / `npm` (deleted the `yarn.lock`). I do not have Yarn v2, so that may be a problem for running using `yarn`

**Comment by sacummings91** [comment]
I'm kinda surprised that this has been such a long ongoing issue. My experience with Tanstack is normally very good but this is becoming pretty annoying. We still use v7 and rely on the docs and now there are none. Is there a way to get this fixed? I'd be happy to help if so.

**Comment by ViieeS** [comment]
Here are all the docs: https://github.com/TanStack/table/tree/v7/docs/src/pages/docs

**Reply by TrinhKhanhTungGITHUB** [reply]
Like

**Comment by KasTasElis** [comment]
Please return the v7 docs. Why would you remove them? 🤦‍♂️

**Reply by Raj00377** [reply]
Yeah absolutely!.

**Comment by leq382121** [comment]
I'm sorry, I know this is Open source, 

But knowing that this tool is being used in big and small corps and knowing that they support this project by donation..

It is extremely selfish to:

1. Have a total rewriting of whole library
2. Change how things are being rendered
3. Remove v7 from site
4. have an EXTREMELY unhelpful migration documentation. 

PLEASE:

1. Put much more effort to migration documentation (v7 -> v8)
2. Explain how things are changing without any laughs and gags - this is serious and it cost money and time. 
3. Have examples of all tree shaking functions to what names they are changing. 

I have to rewrite now the whole implementation that is being used by thousands of users and it is really frustrating to do so because it is very unclear what is changed to what. Sorry about the rant, but it is really selfish to remove v7 documentation from site which means this version will soon depreciate as a result we are FORCED to read and do magic mingling with unclear documentation.

**Reply by dariolongo-dufercodev** [reply]
I totally agree!!

**Comment by radandevist** [comment]
You can build and host the docs of v7 yourself it is just a Next.js static site. It is not so obvious at the beginning but these are the step I did:
1 - clone the v7 branch of the react-tabe
2 - install node v 16.20.2 and use it (you may need nvm)
3 - cd into the docs folder of the cloned repo
4 - npm install inside that folder
5 - run the command `npx next export` (or  `npm run build`)
6- if you used `npm next export`:
    a - install globally http-server-spa
    b run the command `http-server-spa out`
 And here you go!!
 I hosted the output folder online so I don't have to run a command everytime I need to check the docs.

**Reply by Raj00377** [reply]
@radandevist Can you share the hosted link?

**Reply by willycamargo** [reply]
https://react-table-v7-docs.netlify.app

**Comment by willycamargo** [comment, marked as answer]
Hosted it on Netlify for the ones interested:

https://react-table-v7-docs.netlify.app/

**Reply by rhw-repo** [reply]
@willycamargo your link works great for me, thankyou so much, I've used v7 in an app and been just relying on the github docs for a while when I wanted to "go back to find something", so this is an absolute gift, appreciate!

**Reply by iseekTo** [reply]
Yesss!!!! Take me tour 🚀

**Reply by julian47david** [reply]
You are the best!!!

**Reply by kacperklosowski** [reply]
Thank you!

**Reply by daveeeeeeed** [reply]
My man! You rock Willy, thank you so much!!

**Reply by AbhirajSinha179** [reply]
tysm man!

**Reply by ss-joy** [reply]
Thanks man

**Comment by amrendracs** [comment]
https://github.com/TanStack/table/blob/v7/docs/src/pages/docs/api/useTable.md

**Comment by nukeop** [comment]
What a low move to delete the docs in an attempt to force developers to upgrade. They even broke the archive page on purpose.

**Reply by tannerlinsley** [reply]
There was no malicious intent here I assure you. We moved the docs to an old branch and did a rewrite of the docs site. Everything is still accessible. Someone above even stood up the old site. Also, chill. This is free open source software, not a cheap SaaS product.  I’m sure Nuclear is great and likewise you would never want anyone bringing this kind of attitude to your repo.

**Reply by aaryn101** [reply]
@tannerlinsley You did state that you were moving the new docs over back in June of 2023. That never actually happened and you never posted an update.

It is pretty reasonable for developers to be frustrated by this and for them to express their frustration. Telling them to "chill" because it is open source doesn't help anything and only increases their frustration.

**Reply by sydk-adhoc** [reply]
I'd just like to balance this thread by thanking @tannerlinsley for his work. I would like to see the docs updated on Tanstack, and if that can't happen, it seems like it might make sense to have a link or instructions in the official docs about how to get to V7 documentation. @willycamargo (also, thank you), are you planning to maintain your site?
